gateway/gateway.go

```diff
@@ -288,7 +305,20 @@ func (g *gateway) Reconfigure(ctx context.Context, deps resource.Dependencies, c
 		return fmt.Errorf("error initializing the gateway: %w", err)
 	}
 
-	errCode := C.setUpGateway(C.int(cfg.Bus))
+	region := US
+	if strings.Contains(cfg.Region, "EU") || strings.Contains(cfg.Region, "868") {
+		g.logger.Infof("configuring for EU868 region")
+		region = EU
+	}
```

Comment:
reuse the isValidRegion that returns the region code. then if unspecified, set to default/US